orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -503,3 +503,17 @@ def _get_inference_graph(self, *inputs, **kwargs):
                 'There was an error while exporting the PyTorch model to ONNX: {}'.format(e))
 
         return onnx.load_model_from_string(f.getvalue())
+
+    def state_dict(self, destination=None, prefix='', keep_vars=False):
+        # Override the state_dict() method so that the state dict key names
+        # do not contain the _flattened_output_module._base_module prefix
+        return self._flattened_output_module._base_module.state_dict(
+            destination=destination, prefix=prefix, keep_vars=keep_vars)
+
+    def load_state_dict(self, state_dict: 'OrderedDict[str, Tensor]',
+                        strict: bool = True):
+        # Override the load_state_dict() method so that the loaded state dict
+        # key names does not need to contain the _flattened_output_module._base_module prefix
+        return self._flattened_output_module._base_module.load_state_dict(
+            state_dict, strict=strict)
```

Comment:
Thanks for making this change! Are there any other methods that we should override in a similar way, for consistency?

e.g. I could imagine that any method dealing in parameter *names* should be delegated to the base module. So maybe `named_parameters()`, `named_buffers()`, others...?

Reply:
We can start with:
- ```state_dict()```
- ```load_state_dict()```
- ```parameters()```
- ```named_parameters()```
- ```buffers()```
- ```named_buffers()```
for now. And can add support as required.

Reply:
I also added a few more methods:
- ```register_buffer()```
- ```register_parameter()```
- ```get_buffer()```
- ```get_parameter()```